project-koku/koku
# COST-1284: Throw 400 when using start_date/end_date parameters with monthly resolution

**Testing**
1. Verify daily filter resolution works
2. Verify monthly filter resolution throws ValidationError (400)

**Test Results**
1.
```GET /api/cost-management/v1/reports/aws/costs/?start_date=2021-02-01&end_date=2021-02-16&filter[resolution]=daily
HTTP 200 OK
Allow: GET, HEAD, OPTIONS
Content-Type: application/json
Vary: X_RH_IDENTITY, Accept

{
    "meta": {
        "count": 16,
        "start_date": "2021-02-01",
        "end_date": "2021-02-16",
        "compute_count": false,
        "check_tags": false,
        "filter": {
            "resolution": "daily"
        },
        "group_by": {},
        "order_by": {},
        "total": {
            "infrastructure": {


GET /api/cost-management/v1/reports/aws/instance-types/?start_date=2021-02-01&end_date=2021-02-16&filter[resolution]=daily
HTTP 200 OK
Allow: GET, HEAD, OPTIONS
Content-Type: application/json
Vary: X_RH_IDENTITY, Accept

{
    "meta": {
        "count": 16,
        "start_date": "2021-02-01",
        "end_date": "2021-02-16",
        "compute_count": false,
        "check_tags": false,
        "filter": {
            "resolution": "daily"
        },


```

2.

```
GET /api/cost-management/v1/reports/aws/costs/?start_date=2021-02-01&end_date=2021-02-16&filter[resolution]=monthly
HTTP 400 Bad Request
Allow: GET, HEAD, OPTIONS
Content-Type: application/json
Vary: X_RH_IDENTITY, Accept

{
    "error": [
        "Monthly resolution is not supported with start_date and end_date parameters."
    ]
}

GET /api/cost-management/v1/reports/aws/instance-types/?start_date=2021-02-01&end_date=2021-02-16&filter[resolution]=monthly
HTTP 400 Bad Request
Allow: GET, HEAD, OPTIONS
Content-Type: application/json
Vary: X_RH_IDENTITY, Accept

{
    "error": [
        "Monthly resolution is not supported with start_date and end_date parameters."
    ]
}

GET /api/cost-management/v1/reports/openshift/costs/?start_date=2021-02-01&end_date=2021-02-16&filter[resolution]=monthly
HTTP 400 Bad Request
Allow: GET, HEAD, OPTIONS
Content-Type: application/json
Vary: X_RH_IDENTITY, Accept

{
    "error": [
        "Monthly resolution is not supported with start_date and end_date parameters."
    ]
}

GET /api/cost-management/v1/reports/azure/costs/?start_date=2021-02-01&end_date=2021-02-16&filter[resolution]=monthly
HTTP 400 Bad Request
Allow: GET, HEAD, OPTIONS
Content-Type: application/json
Vary: X_RH_IDENTITY, Accept

{
    "error": [
        "Monthly resolution is not supported with start_date and end_date parameters."
    ]
}

GET /api/cost-management/v1/reports/gcp/costs/?start_date=2021-02-01&end_date=2021-02-16&filter[resolution]=monthly
HTTP 400 Bad Request
Allow: GET, HEAD, OPTIONS
Content-Type: application/json
Vary: X_RH_IDENTITY, Accept

{
    "error": [
        "Monthly resolution is not supported with start_date and end_date parameters."
    ]
}



```
